tests/pylint/test_enforce_greek_micro_char.py

```diff
@@ -0,0 +1,160 @@
+from pylint.utils.ast_walker import ASTWalker
+import pytest
+
+from . import assert_no_messages
+
+
+@pytest.mark.parametrize(
+    "code",
+    [
+        pytest.param(
+            # Test using the correct μ-sign \u03bc with annotation
+            """
+        CONCENTRATION_MICROGRAMS_PER_CUBIC_METER: Final = "μg/m³"
+        """,
+            id="good_const_with_annotation",
+        ),
+        pytest.param(
+            # Test using the correct μ-sign \u03bc with annotation using unicode encoding
+            """
+        CONCENTRATION_MICROGRAMS_PER_CUBIC_METER: Final = "\u03bcg/m³"
+        """,
+            id="good_unicode_const_with_annotation",
+        ),
+        pytest.param(
+            # Test using the correct μ-sign \u03bc without annotation
+            """
+        CONCENTRATION_MICROGRAMS_PER_CUBIC_METER = "μg/m³"
+        """,
+            id="good_const_without_annotation",
+        ),
+        pytest.param(
+            # Test using the correct μ-sign \u03bc in a StrEnum class
+            """
+            class UnitOfElectricPotential(StrEnum):
+                \"\"\"Electric potential units.\"\"\"
+
+                MICROVOLT = "μV"
+                MILLIVOLT = "mV"
+                VOLT = "V"
+                KILOVOLT = "kV"
+                MEGAVOLT = "MV"
+        """,
+            id="good_str_enum",
+        ),
+        pytest.param(
+            # Test using the correct μ-sign \u03bc in a sensor description dict
+            """
+            SENSOR_DESCRIPTION = {
+                "radiation_rate": AranetSensorEntityDescription(
+                    key="radiation_rate",
+                    translation_key="radiation_rate",
+                    name="Radiation Dose Rate",
+                    native_unit_of_measurement="μSv/h",
+                    state_class=SensorStateClass.MEASUREMENT,
+                    suggested_display_precision=2,
+                    scale=0.001,
+                ),
+            }
+            OTHER_DICT = {
+                "value_with_bad_mu_should_pass": "µ"
+            }
+        """,
+            id="good_sensor_description",
+        ),
+    ],
+)
+def test_enforce_greek_micro_char(
+    linter: UnittestLinter,
+    enforce_greek_micro_char_checker: BaseChecker,
+    code: str,
+) -> None:
+    """Good test cases."""
+    root_node = astroid.parse(code, "homeassistant.components.pylint_test")
+    walker = ASTWalker(linter)
+    walker.add_checker(enforce_greek_micro_char_checker)
+
+    with assert_no_messages(linter):
+        walker.walk(root_node)
+
+
+@pytest.mark.parametrize(
+    "code",
+    [
+        pytest.param(
+            # Test using the bad μ-sign \u00b5 instead of \u03bc with annotation
+            """
+            CONCENTRATION_MICROGRAMS_PER_CUBIC_METER: Final = "µg/m³"
+        """,
+            id="bad_const_with_annotation",
+        ),
+        pytest.param(
+            # Test using the bad μ-sign \u00b5 instead of \u03bc with unicode literal
```

Comment:
I think this comment is weird, it's not bad to code µ as \u00b5, we've just decided to use \u03bc.
Maybe use a language similar to the comments in number.const / sensor.const?